File: generated-code/codecraft/java/src/main/java/model/BuildProperties.java
package model;

import util.StreamUtil;

public class BuildProperties {
    private model.EntityType[] options;

    public model.EntityType[] getOptions() {
        return options;
    }

    public void setOptions(model.EntityType[] value) {
        this.options = value;
    }
    private Integer initHealth;

    public Integer getInitHealth() {
        return initHealth;
    }

    public void setInitHealth(Integer value) {
        this.initHealth = value;
    }

    public BuildProperties(model.EntityType[] options, Integer initHealth) {
        this.options = options;
        this.initHealth = initHealth;
    }

    public static BuildProperties readFrom(java.io.InputStream stream) throws java.io.IOException {
        model.EntityType[] options;
        options = new model.EntityType[StreamUtil.readInt(stream)];
        for (int optionsIndex = 0; optionsIndex < options.length; optionsIndex++) {
            model.EntityType optionsElement;
            optionsElement = model.EntityType.readFrom(stream);
            options[optionsIndex] = optionsElement;
        }
        Integer initHealth;
        if (StreamUtil.readBoolean(stream)) {
            initHealth = StreamUtil.readInt(stream);
        } else {
            initHealth = null;
        }
        return new BuildProperties(options, initHealth);
    }

    public void writeTo(java.io.OutputStream stream) throws java.io.IOException {
        StreamUtil.writeInt(stream, options.length);
        for (model.EntityType optionsElement : options) {
            StreamUtil.writeInt(stream, optionsElement.tag);
        }
        if (initHealth == null) {
            StreamUtil.writeBoolean(stream, false);
        } else {
            StreamUtil.writeBoolean(stream, true);
            StreamUtil.writeInt(stream, initHealth);
        }
    }

    @Override
    public String toString() {
        StringBuilder stringBuilder = new StringBuilder("BuildProperties { ");
        stringBuilder.append("options: ");
        stringBuilder.append("[ ");
        for (int optionsIndex = 0; optionsIndex < options.length; optionsIndex++) {
            if (optionsIndex != 0) {
                stringBuilder.append(", ");
            }
            model.EntityType optionsElement = options[optionsIndex];
            stringBuilder.append(String.valueOf(optionsElement));
        }
        stringBuilder.append(" ]");
        stringBuilder.append(", ");
        stringBuilder.append("initHealth: ");
        stringBuilder.append(String.valueOf(initHealth));
        stringBuilder.append(" }");
        return stringBuilder.toString();
    }
}

File: generated-code/codecraft/kotlin/src/main/kotlin/model/BuildProperties.kt
package model

import util.StreamUtil

class BuildProperties {
    lateinit var options: Array<model.EntityType>
    var initHealth: Int? = null

    constructor(options: Array<model.EntityType>, initHealth: Int?) {
        this.options = options
        this.initHealth = initHealth
    }

    @Throws(java.io.IOException::class)
    fun writeTo(stream: java.io.OutputStream) {
        StreamUtil.writeInt(stream, options.size)
        for (optionsElement in options) {
            StreamUtil.writeInt(stream, optionsElement.tag)
        }
        val initHealthValue = initHealth
        if (initHealthValue == null) {
            StreamUtil.writeBoolean(stream, false)
        } else {
            StreamUtil.writeBoolean(stream, true)
            StreamUtil.writeInt(stream, initHealthValue)
        }
    }

    override fun toString(): String {
        var stringBuilder = StringBuilder("BuildProperties { ")
        stringBuilder.append("options: ")
        stringBuilder.append("[ ")
        var optionsIndex = 0
        for (optionsElement in options) {
            if (optionsIndex != 0) {
                stringBuilder.append(", ")
            }
            stringBuilder.append(optionsElement)
            optionsIndex++
        }
        stringBuilder.append(" ]")
        stringBuilder.append(", ")
        stringBuilder.append("initHealth: ")
        stringBuilder.append(initHealth)
        stringBuilder.append(" }")
        return stringBuilder.toString()
    }

    companion object {
        @Throws(java.io.IOException::class)
        fun readFrom(stream: java.io.InputStream): BuildProperties {
            var options: Array<model.EntityType>
            options = Array(StreamUtil.readInt(stream), {
                var optionsElement: model.EntityType
                optionsElement = model.EntityType.readFrom(stream)
                optionsElement
            })
            var initHealth: Int?
            if (StreamUtil.readBoolean(stream)) {
                initHealth = StreamUtil.readInt(stream)
            } else {
                initHealth = null
            }
            return BuildProperties(options, initHealth)
        }
    }
}